### PR TITLE
Use a private variable to cache the default style array

### LINKED ISSUE #4063: ol.style.defaultStyleFunction lazy create not working
See #1815.
Code https://github.com/openlayers/ol3/blob/master/src/ol/style/style.js#L267-L281

The replaced function is never called because `ol.layer.Vector` stores a reference to this function before it is replaced.

### PR DESCRIPTION
**Pull request by fredj**
See #1815
Fixes #4063

### CODE REVIEW

**Comment by gberaudo** [comment]
+1

File: src/ol/style/style.js [hunk 1]
```diff
@@ -240,47 +240,39 @@ ol.style.createStyleFunction = function(obj) {
...
-      stroke: stroke
-    })
-  ];
-
-  // Now that we've run it the first time, replace the function with
-  // a constant version. We don't use an immediately-invoked function
-  // and a closure so we don't get an error at script evaluation time in
-  // browsers that do not support Canvas. (ol.style.Circle does
-  // canvas.getContext('2d') at construction time, which will cause an.error
-  // in such browsers.)
```
Comment (elemoine) [inline src/ol/style/style.js]: I just think that we should leave that comment, or a similar comment, to explain why we don't want to eagerly create the default styles.

**Comment by elemoine** [comment]
I just added a comment. Please merge when addressed.

**Comment by fredj** [comment]
comment addressed, thanks for the review

**Pull request merged**